File: acmicpc/acmicpc10773.cpp
#include <stdio.h>
#include <stack>

int main(){
    std::stack<int> arr;
    unsigned int n;
    scanf("%d", &n);
    unsigned int temp;
    for (int i=0; i < n; i++){
        scanf("%d", &temp);
        if (temp == 0){
            arr.pop();
        } else {
            arr.push(temp);
        }
    }
    long long int sum = 0;
    int size = arr.size();
    for (int i=0; i < size; i++){
        sum += arr.top();
        arr.pop();
    }
    printf("%lld\n", sum);
}

File: acmicpc/acmicpc10773.py
import time
import random

t0 = time.time()

# n = 100000
# arr = []
# for _ in range(n):
#     temp = random.randint(0, 100)
#     if (temp == 0) and (len(arr) != 0):
#         arr = arr[0:len(arr)-1]
#     else:
#         arr.append(temp)

import sys

n = int(sys.stdin.readline())
arr = []
for _ in range(n):
    temp = int(sys.stdin.readline())
    if (temp == 0) and (len(arr) != 0):
        arr = arr[0:len(arr)-1]
    else:
        arr.append(temp)

print(sum(arr))

t1 = time.time()
print('elapsed: {}'.format(t1 - t0))